crates/bevy_reflect/src/list.rs

```diff
@@ -522,4 +524,23 @@ mod tests {
             assert_eq!(index, value);
         }
     }
+
+    #[cfg(not(debug_assertions))]
```

Comment:
Does CI run tests in release mode? If not, it might be something we raise an issue for such that tests like these can be run.

Reply:
[it doesn't](https://github.com/bevyengine/bevy/blob/0dddfa07ab5bdb95b0dcc1041e423678f92e83f7/tools/ci/src/commands/test.rs#L20)...